File: NumMeth/Demos/counting/CountingFloat.java

public class CountingFloat {

	static void showval(float f) {
		int n = Float.floatToIntBits(f);
		System.out.printf("%.8e (0x%08x or %d:%06x)\n", 
				f, n, n>>23 & 0xff, n & 0x7fffff);
	}
	
	/**
	 * @param args
	 */
	public static void main(String[] args) {
		float i, j;
		i = 0;
		do {
			j = i;
			i++;
		} while (i != j);
		showval(i);
	}

}

File: Part IA/NumMeth/Demos/counting/counting_float.c
#include <stdio.h>

/* what happens when we increment a 'float' forever...                  */

static void showval(float f)
{   union { float f; int n; } u;
    u.f = f;
    /* This use of 'union' is *not* portable, but it works on x86/gcc.  */
    printf("%.8e (0x%.8x or %d:%.6x)\n", f, u.n,
           u.n>>23 & 0xff, u.n & 0x7fffff);
}

int main()
{ float i=0,j;
  do { j=i; i++; } while (i != j);
  showval(i);
  return 0;
}

//eof
